securedrop/journalist_app/api2/events.py

```diff
@@ -135,3 +128,160 @@ def handle_reply_sent(event: Event) -> EventResult:
             sources={source.uuid: source},
             items={reply.uuid: reply},
         )
+
+    @staticmethod
+    def handle_source_deleted(event: Event) -> EventResult:
+        from journalist_app.api2 import json_version  # cyclic import
+
+        try:
+            source = Source.query.filter(Source.uuid == event.target.source_uuid).one()
+        except NoResultFound:
+            return EventResult(
+                event_id=event.id,
+                status=(
+                    EventStatusCode.NotFound,
+                    f"could not find source: {event.target.source_uuid}",
+                ),
+            )
+
+        current_version = json_version(source.to_api_v2())
+        if event.target.version != current_version:
+            return EventResult(
+                event_id=event.id,
+                status=(
+                    EventStatusCode.BadRequest,
```

Comment:
Might be a little richer semantically to use [HTTP `409 Conflict`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status/409):

```suggestion
                    EventStatusCode.Conflict,
```